dump/mapping/bufr_ssmis.py

```diff
@@ -0,0 +1,241 @@
+
+    try:
+        return int(env_nprocs or slurm_nprocs or default)
+    except ValueError:
+        print(f"[WARN] Invalid environment variable value. Falling back to default: {default}")
+        return default
+
+
+def compute_solar_angles(lat, lon, unix_time):
+    """
+    Compute solar zenith and azimuth angles for a single point.
+
+    Parameters:
+        lat: Latitude in degrees.
+        lon: Longitude in degrees.
+        unix_time: Unix timestamp (seconds since 1970-01-01T00:00:00Z).
+    Returns:
+        tuple: zenith angle, azimuth angle
+    """
+
+    dt = datetime.fromtimestamp(int(unix_time), tz=timezone.utc)
+    altitude = get_altitude(lat, lon, dt)
+    azimuth = get_azimuth(lat, lon, dt)
+    zenith = 90.0 - altitude
+
+    return zenith, azimuth
+
+
+MAPPING_PATH = map_path('bufr_ssmis.yaml')
+
+
+class BufrSsmisObsBuilder(ObsBuilder):
+    def __init__(self):
+        super().__init__(MAPPING_PATH, log_name=os.path.basename(__file__))
+
+    def make_obs(self, comm, input_path):
+        # Get container from mapping file first
+        self.log.info('Get container from bufr')
+        container = super().make_obs(comm, input_path)
+
+        self.log.debug(f'container list (original): {container.list()}')
+        self.log.debug(f'all_sub_categories =  {container.all_sub_categories()}')
+        self.log.debug(f'category map =  {container.get_category_map()}')
+
+        # Add new/derived data into container
+        for cat in container.all_sub_categories():
+
+            self.log.debug(f'category = {cat}')
+
+            satId = container.get('satelliteId', cat)
+            if not np.any(satId):
+                self.log.warning(f'category {cat[0]} does not exist in input file')
+
+            self._add_solar_angles(container, cat)
+            self._add_satellite_ascend_descent_orbit(container, cat)
+
+        # Check
+        self.log.debug(f'container list (updated): {container.list()}')
+        self.log.debug(f'all_sub_categories {container.all_sub_categories()}')
+
+        return container
+
+    def _make_description(self):
+        description = super()._make_description()
+        self._add_solar_angles_descriptions(description)
+        self._add_satellite_ascend_descend_orbit_descriptions(description)
+
+        return description
+
+    def _add_solar_angles_descriptions(self, description):
+        description.add_variables([
+            {
+                'name': 'MetaData/solarZenithAngle',
+                'source': 'solarZenithAngle',
+                'units': 'degree',
+                'longName': 'Solar Zenith Angle',
+            },
+            {
+                'name': 'MetaData/solarAzimuthAngle',
+                'source': 'solarAzimuthAngle',
+                'units': 'degree',
+                'longName': 'Solar Azimuth Angle',
+            }])
+
+    def _add_satellite_ascend_descend_orbit_descriptions(self, description):
```

Comment:
Can we merge this with ```_add_solar_angles_descriptions``` above?

Reply:
Yes.   I combined the addition of new variable descriptions into one.